saged/datasets.py

```diff
@@ -320,34 +366,20 @@ def __len__(self) -> int:
         """
         return len(self.current_expression.columns)
 
-    def __getitem__(self, idx: int) -> Tuple[np.array, np.array]:
+    def set_all_data(self, new_data: np.array) -> None:
         """
-        Allows access into the dataset to select a single datapoint
+        Overwrite the current data stored in the dataset with the new one passed in
 
         Arguments
         ---------
-        idx: The index of the given item
-
-        Returns
-        -------
-        sample: The gene expression data for the given index in a genes x 1 array
-        """
-        sample_id = self.get_samples()[idx]
-        sample = self.current_expression[sample_id].values
-        return sample
-
-    def get_all_data(self) -> Tuple[np.array, np.array]:
+        new_data: The variable x samples array to overwrite the old data with. The first dimension
+                  can be anything, but the sample dimension must remain the same as in the old data
```

Comment:
It's not just that the dimension has to be the same, but the columns have to be in the same order too, right?

Reply:
That's true, updated the comment